app/classifier/classification-summary.jsx

```diff
@@ -1,54 +1,89 @@
-      const SummaryComponent = tasks[task.type].Summary; // TODO: There's a lot of duplicated code in these modules.
-      body.push(
-        <div key={annotation._key} className="classification-task-summary">
-          <SummaryComponent task={task} annotation={annotation} onToggle={props.onToggle} />
-        </div>
+
+  render() {
+    const tools = this.props.project.experimental_tools || [];
+
+    if (tools.includes('metadata-based-feedback')) {
+      return (
+        <MetadataBasedFeedback
+          subject={this.props.subject}
+          classification={this.props.classification}
+          dudLabel="DUD"
+          simLabel="SIM"
+          subjectLabel="SUB"
+          metaTypeFieldName="#Type"
+          metaSuccessMessageFieldName="#F_Success"
+          metaFailureMessageFieldName="#F_Fail"
+          metaSimCoordXPattern="#X"
+          metaSimCoordYPattern="#Y"
+          metaSimTolPattern="#Tol"
+        />
       );
-    });
-  }
-  return (
-    <div className="classification-summary">
-      {firstTimeClassified}
-      {body}
-    </div>
-  );
-};
+    }
 
-ClassificationSummary.defaultProps = {
-  workflow: null,
-  classification: null,
-  classificationCount: null
-};
+    if (tools.includes('world_wide_telescope')) {
+      return (
+        <strong>
+          <WorldWideTelescope
+            annotations={this.props.classification.annotations}
+            subject={this.props.subject}
+            workflow={this.props.workflow}
+          />
+        </strong>
+      );
+    }
+
+    return (
+      <div>
+        { this.hasExpert ?
+          <div className="has-expert-classification">
+            Expert classification available.&nbsp;
+            { this.state.showExpert ?
+              <button type="button" onClick={this.dontShowExpert}>Hide</button> :
+              <button type="button" onClick={this.doShowExpert}>Show</button> }
+          </div> : '' }
+
+        <div>
+          <strong>
+            { this.state.showExpert ? 'Expert Classification:' : 'Your classification:' }
+          </strong>
+          <DefaultClassificationSummary
+            workflow={this.props.workflow}
+            classification={this.state.showExpert ? this.props.expertClassification : this.props.classification}
+            classificationCount={this.props.classificationCount}
+            splits={this.props.splits}
+          />
+        </div>
+      </div>
+    );
+  }
+}
 
 ClassificationSummary.propTypes = {
-  workflow: React.PropTypes.object,
-  classification: React.PropTypes.object,
-  classificationCount: React.PropTypes.number,
-  onToggle: React.PropTypes.func,
-  splits: React.PropTypes.object
+  project: React.PropTypes.object.isRequired,
+  workflow: React.PropTypes.object.isRequired,
+  subject: React.PropTypes.object.isRequired,
+  classification: React.PropTypes.object.isRequired,
+  expertClassification: React.PropTypes.object,
+  splits: React.PropTypes.array,
```

Comment:
I'm not sure why this changed, but this now has a prop type validation error. It should be an object.